ad-code-manager.php

```diff
@@ -272,52 +272,52 @@ function update_options( $new_options ) {
 	 */
 	function handle_admin_action() {
 
-		if ( !wp_verify_nonce( $_REQUEST['nonce'], 'acm-admin-action' ) )
-			wp_die( __( 'Doing something fishy, eh?', 'ad-code-manager' ) );
+		if ( isset( $_REQUEST['nonce'] ) && ! wp_verify_nonce( sanitize_key( $_REQUEST['nonce'] ), 'acm-admin-action' ) ) // Input var okay.
+			wp_die( esc_html( __( 'Doing something fishy, eh?', 'ad-code-manager' ) ) );
 
 		if ( !current_user_can( $this->manage_ads_cap ) )
-			wp_die( __( 'You do not have the necessary permissions to perform this action', 'ad-code-manager' ) );
+			wp_die( esc_html( __( 'You do not have the necessary permissions to perform this action', 'ad-code-manager' ) ) );
 
 		// Depending on the method we're performing, sanitize the requisite data and do it
-		switch ( $_REQUEST['method'] ) {
+		switch ( sanitize_text_field( $_REQUEST['method'] ) ) { // Input var okay.
 		case 'add':
 		case 'edit':
-			$id = ( isset( $_REQUEST['id'] ) ) ? (int)$_REQUEST['id'] : 0;
-			$priority = ( isset( $_REQUEST['priority'] ) ) ? (int)$_REQUEST['priority'] : 10;
-			$operator = ( isset( $_REQUEST['operator'] ) && in_array( $_REQUEST['operator'], array( 'AND', 'OR' ) ) ) ? $_REQUEST['operator'] : $this->logical_operator;
+			$id = ( isset( $_REQUEST['id'] ) ) ? absint( $_REQUEST['id'] ) : 0; // Input var okay.
+			$priority = ( isset( $_REQUEST['priority'] ) ) ? absint( $_REQUEST['priority'] ) : 10; // Input var okay.
+			$operator = ( isset( $_REQUEST['operator'] ) && in_array( $_REQUEST['operator'], array( 'AND', 'OR' ) ) ) ? sanitize_text_field( $_REQUEST['operator'] ) : $this->logical_operator; // Input var okay.
```

Comment:
in_array() should have the third param set to true to test for equality and not just PHP's "close enough"